### PR TITLE
fix: ledger mapper validity interval prop name

### PR DESCRIPTION
**Pull request by mkazlauskas**
# Context

Ledger was signing a different transaction hash due to incorrect property name when mapping transaction props

# Proposed Solution

Fix it to the correct property name and also add a check to throw an error if Ledger signs a different transaction hash

# Important Changes Introduced

A few chores that fix minor issues with SDK setup:
- 389f8d4eff9c29543ff4a14c282ec87107635bd7
- 5a96d0b2f0829159b0046848cb8c7473cbdccfdd

### CODE REVIEW

**Review by rhyslbw — DISMISSED** [review]
Nice work @mkazlauskas. Great improvement to verifying the hardware result to fail early

**Review by mirceahasegan — DISMISSED** [review]
Nice work!

File: packages/key-management/src/LedgerKeyAgent.ts [hunk 1]
```diff
@@ -239,6 +239,9 @@ export class LedgerKeyAgent extends KeyAgentBase {
         this.deviceConnection
       );
       const result = await deviceConnection.signTransaction(ledgerTxData);
+      if (result.txHashHex !== hash.toString()) {
```
Comment (iadmytro) [inline packages/key-management/src/LedgerKeyAgent.ts:242]: It would be good to add test for this case.

Comment (rhyslbw) [reply packages/key-management/src/LedgerKeyAgent.ts:242]: It's a tricky setup in a unit test, for not that much value, as this exception will be covered by integration tests that demonstrate we are able to construct transactions for each property.

Comment (mkazlauskas) [reply packages/key-management/src/LedgerKeyAgent.ts:242]: Added a test in 2ba7be9
Also fixed existing test, it was failing due to using different (mocked) output addresses and producing a different transaction hash

**Review by rhyslbw — APPROVED** [review]

**Review by iadmytro — DISMISSED** [review]

**Pull request merged**